### ISSUE TITLE
Reorganize OrcExamples

### ISSUE THREAD

**Issue by jthywiss** [orc-lang/orc#197]
The structure of OrcExamples has become helter-skelter.  This is bad for a new user trying to find some examples to learn Orc by.

For example, there's an "applications" directory.  Aren't all of the examples applications?  Some directories are named after user goals (beginning_Orc), some after publications (orco_paper), some after language features (objects), and some after types of program (distributed_algorithms).  A mess.

Reorganize cleanly in user-goal-focused structure.

**Comment by arthurp** [comment]
I agree, but I think we need to talk about the specific goal of organization.

The intention with the directories in my mind is to categorize the intent of the example, not it's content. What I mean is that an example that should teach you about objects should be in the objects/ directory. But if it simply uses objects, but actually teaches about distributed programming it would be in a distributed/ directory.

I agree that the applications directory is a problem, however I disagree that all examples are applications. Most are very small and are programs but not applications; the word application implies a larger scale to me than example or program. I created the applications/ directory to hold larger applications which focus on the application of Orc instead of the specific Orc features used to implement the program.
